client/src/pages/Profile.js

```diff
@@ -78,26 +88,53 @@ const Profile = () => {
   };
 
   const onAttributeChange = (value, attributeLabel) => {
-    setUser({
-      ...user,
+    setLocalUser({
+      ...localUser,
       [attributeLabel]: value,
     });
   };
 
+  const createUpdatedUser = () => {
+    const updatedUser = {};
+    userPermissions.edit.forEach((field) => {
+      updatedUser[field] = localUser[field];
+    });
+
+    return updatedUser;
+  };
+
+  const setTemporarySuccessMessage = (contents) => {
+    setSuccessMessage(contents);
+    setTimeout(() => setSuccessMessage(null), SUCCESS_MESSAGE_POPUP_TIME_MS);
+  };
+
+  const submitChanges = async () => {
+    const result = await updateMember(createUpdatedUser(), upstreamUser._id);
+    if (!areResponsesSuccessful(result)) {
+      setErrorMessage(
+        `An error occured${
+          result && result.data && result.data.message
+            ? `: ${result.data.message}`
+            : '.'
+        }`,
+      );
+      setSuccessMessage(null);
+    } else {
+      setTemporarySuccessMessage('User updated');
+      setErrorMessage(null);
```

Comment:
Last suggestion is to reupdate UpstreamUser here with `result.data.result` (which contains the updated user) so the button gets disabled again.